### PR TITLE
[BD-7321][BpkMap] change icon marker style(color)

### PR DESCRIPTION
**Pull request by LuoTuxiu**
The style of Component `BpkIconMarker` changed

| Before                        | After                         |
| ----------------------------- | ----------------------------- |
| image | image |

## Figma

https://www.figma.com/file/kOzCI1zFdIOcZGkZr2vHTR/Map-Redesign--(Google-Map%2C-Pins%2C-POI)?type=design&node-id=488-14862&mode=design&t=q14DEHsRfsxPVLRK-0

## Slack

https://skyscanner.slack.com/archives/C05PK45MWTG/p1695645460916799

Remember to include the following changes:

- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss [hunk 1]
```diff
@@ -26,7 +26,7 @@
   @include bpk-themeable-property(
     fill,
     --bpk-icon-marker-default-background-color,
-    $bpk-status-success-spot-day
+    $bpk-marcomms-purple
```
Comment (jasonhuang-sky) [inline packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss:29]: It's likely a deprecated color token, are you sure it's correct?

File: packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss [hunk 2]
```diff
@@ -42,12 +42,12 @@
   }
 
   &--selected {
-    fill: $bpk-surface-default-day;
+    fill: $bpk-marcomms-dark-sky;
```
Comment (jasonhuang-sky) [inline packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss:45]: same this one

File: packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss [hunk 3]
```diff
@@ -42,12 +42,12 @@
...
 
     @include bpk-themeable-property(
       stroke,
       --bpk-icon-marker-default-selected-color,
-      $bpk-status-success-spot-day
+      $bpk-marcomms-dark-sky
```
Comment (jasonhuang-sky) [inline packages/bpk-component-map/src/BpkIconMarkerBackground.module.scss:50]: same this one

**Review by jasonhuang-sky — APPROVED** [review]

**Review by olliecurtis — CHANGES_REQUESTED** [review]
I notice that the disabled version of the icon marker is still marked as green. Could you check if this is expected or not?

https://github.com/Skyscanner/backpack/pull/3028/files#diff-910dcd001032877120cf36bfb367b3b519423a2e451cee375701ed560386ed3cR82

Screenshot 2023-10-09 at 09 43 45

**Comment by LuoTuxiu** [comment]
> I notice that the disabled version of the icon marker is still marked as green. Could you check if this is expected or not?
> 
> https://github.com/Skyscanner/backpack/pull/3028/files#diff-910dcd001032877120cf36bfb367b3b519423a2e451cee375701ed560386ed3cR82
> 
> Screenshot 2023-10-09 at 09 43 45

Thanks. @olliecurtis i think we just focus on changing the color which is default and selected. And we have **not use disabled status**  at our repo for now. So maybe i should confirm with our disigner?

**Comment by anambl** [comment]
> > I notice that the disabled version of the icon marker is still marked as green. Could you check if this is expected or not?
> > https://github.com/Skyscanner/backpack/pull/3028/files#diff-910dcd001032877120cf36bfb367b3b519423a2e451cee375701ed560386ed3cR82
> > Screenshot 2023-10-09 at 09 43 45
> 
> Thanks. @olliecurtis i think we just focus on changing the color which is default and selected. And we have **not use disabled status** at our repo for now. So maybe i should confirm with our disigner?

I think we should be updating the disabled status one as well as this is changing the main icon marker component and is traveller facing so keeping both colours will be confusing and inconsistent.

**Comment by anambl** [comment]
@LuoTuxiu just checking - is this change meant to be stable or will there be an experiment running to test it?

**Comment by LuoTuxiu** [comment]
> @LuoTuxiu just checking - is this change meant to be stable or will there be an experiment running to test it?

@olliecurtis  i confrim with our PO, this will release directly.

File: examples/bpk-component-map/examples.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ import { withRtlSupport } from '../../packages/bpk-component-icon';
 import LandmarkIconSm from '../../packages/bpk-component-icon/sm/landmark';
 import AirportsIconSm from '../../packages/bpk-component-icon/sm/airports';
 import FoodIconSm from '../../packages/bpk-component-icon/sm/food';
-import ParkingIconSm from '../../packages/bpk-component-icon/sm/parking';
+// import ParkingIconSm from '../../packages/bpk-component-icon/sm/parking';
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:30]: Can this be removed?

File: examples/bpk-component-map/examples.js [hunk 5]
```diff
@@ -45,7 +45,7 @@ const AlignedHotelIconSm = withRtlSupport(HotelIconSm);
 const AlignedLandmarkIconSm = withRtlSupport(LandmarkIconSm);
 const AlignedAirportsIconSm = withRtlSupport(AirportsIconSm);
 const AlignedFoodIconSm = withRtlSupport(FoodIconSm);
-const AlignedParkingIconSm = withRtlSupport(ParkingIconSm);
+// const AlignedParkingIconSm = withRtlSupport(ParkingIconSm);
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:48]: Same here

File: examples/bpk-component-map/examples.js [hunk 6]
```diff
@@ -78,7 +78,7 @@ const venues = [
     latitude: 55.9469995,
     longitude: -3.1905666,
     price: '£48',
-    disabled: false,
+    // disabled: false,
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:81]: Same here

File: examples/bpk-component-map/examples.js [hunk 7]
```diff
@@ -88,7 +88,7 @@ const venues = [
     latitude: 55.9439643,
     longitude: -3.1938768,
     price: '£151',
-    disabled: false,
+    // disabled: false,
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:91]: Same here

File: examples/bpk-component-map/examples.js [hunk 8]
```diff
@@ -98,7 +98,7 @@ const venues = [
     latitude: 55.9432205,
     longitude: -3.1955874,
     price: '£62',
-    disabled: false,
+    // disabled: false,
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:101]: Same here

File: examples/bpk-component-map/examples.js [hunk 9]
```diff
@@ -108,20 +108,20 @@ const venues = [
     latitude: 55.9450573,
     longitude: -3.1996687,
     price: '£342',
-    disabled: false,
+    // disabled: false,
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:111]: Same here

File: examples/bpk-component-map/examples.js [hunk 10]
```diff
@@ -108,20 +108,20 @@ const venues = [
...
+  // {
+  //   id: '5',
+  //   name: 'Kolkata Springs Hotel',
+  //   latitude: 55.943621,
+  //   longitude: -3.192098,
+  //   price: 'Sold out',
+  //   disabled: true,
+  //   icon: <AlignedParkingIconSm />,
+  //   airportsIcon: <AlignedAirportsIconSm />,
+  // },
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:124]: Can this be removed?

File: examples/bpk-component-map/examples.js [hunk 11]
```diff
@@ -278,13 +278,14 @@ class StatefulBpkIconMarker extends Component<
...
           <BpkIconMarker
+            key={venue.id}
             position={{ latitude: venue.latitude, longitude: venue.longitude }}
             onClick={() => {
               this.props.action();
               this.selectVenue(venue.id);
             }}
             icon={venue.icon}
-            disabled={venue.disabled}
+            // disabled={venue.disabled}
```
Comment (olliecurtis) [inline examples/bpk-component-map/examples.js:288]: Can this be removed?

File: packages/bpk-component-map/src/BpkIconMarker-test.js [hunk 12]
```diff
@@ -53,19 +53,19 @@ describe('BpkIconMarker', () => {
...
-      <BpkIconMarker position={position} icon={icon} selected disabled />,
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+  // it('should render correctly with both a "selected" and "disabled" attribute (disabled takes precedence)', () => {
+  //   const { asFragment } = render(
+  //     <BpkIconMarker position={position} icon={icon} selected disabled />,
+  //   );
+  //   expect(asFragment()).toMatchSnapshot();
+  // });
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker-test.js:68]: Can this be removed?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 13]
```diff
@@ -33,7 +33,7 @@ const getClassName = cssModules(STYLES);
 export type Props = {
   icon: Node,
   position: LatLong,
-  disabled: boolean,
+  // disabled: boolean,
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:36]: Same here

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 14]
```diff
@@ -44,7 +44,7 @@ const BpkIconMarker = (props: Props) => {
   const {
     buttonProps,
     className,
-    disabled,
+    // disabled,
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:47]: Same here

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 15]
```diff
@@ -60,8 +60,10 @@ const BpkIconMarker = (props: Props) => {
 
   const iconClassNames = getClassName(
     'bpk-icon-marker__icon',
-    onClick && !disabled && 'bpk-icon-marker__icon--interactive',
-    disabled && 'bpk-icon-marker__icon--disabled',
+    onClick &&
+      // && !disabled
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:64]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 16]
```diff
@@ -60,8 +60,10 @@ const BpkIconMarker = (props: Props) => {
...
+      'bpk-icon-marker__icon--interactive',
+    // disabled && 'bpk-icon-marker__icon--disabled',
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:66]: Same here

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 17]
```diff
@@ -73,12 +75,15 @@ const BpkIconMarker = (props: Props) => {
         type="button"
         className={wrapperClassNames}
         onClick={onClick}
-        disabled={disabled}
+        // disabled={disabled}
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:78]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 18]
```diff
@@ -73,12 +75,15 @@ const BpkIconMarker = (props: Props) => {
...
         {...buttonProps}
       >
         <BpkIconMarkerBackground
-          disabled={disabled}
-          interactive={onClick !== null && !disabled}
+          // disabled={disabled}
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:82]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 19]
```diff
@@ -73,12 +75,15 @@ const BpkIconMarker = (props: Props) => {
...
+          interactive={
+            onClick !== null
+            // && !disabled
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:85]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 20]
```diff
@@ -91,15 +96,15 @@ BpkIconMarker.propTypes = {
   icon: PropTypes.node.isRequired,
   position: LatLongPropType.isRequired,
   className: PropTypes.string,
-  disabled: PropTypes.bool,
+  // disabled: PropTypes.bool,
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:99]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.js [hunk 21]
```diff
@@ -91,15 +96,15 @@ BpkIconMarker.propTypes = {
...
   onClick: PropTypes.func,
   selected: PropTypes.bool,
   buttonProps: PropTypes.object, // eslint-disable-line react/forbid-prop-types
 };
 
 BpkIconMarker.defaultProps = {
   className: null,
-  disabled: false,
+  // disabled: false,
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.js:107]: Can you remove this?

File: packages/bpk-component-map/src/BpkIconMarker.module.scss [hunk 22]
```diff
@@ -75,13 +75,13 @@
...
-        $bpk-status-success-spot-day
-      );
-    }
+    // &--disabled {
+    //   @include bpk-themeable-property(
+    //     color,
+    //     --bpk-icon-marker-default-disabled-color,
+    //     $bpk-status-success-spot-day
+    //   );
+    // }
```
Comment (olliecurtis) [inline packages/bpk-component-map/src/BpkIconMarker.module.scss:84]: Can you remove this?

**Review by olliecurtis — CHANGES_REQUESTED** [review]

**Comment by LuoTuxiu** [comment]
Hi， @olliecurtis  i have updated all deleted annotation, could you help to reivew? Thanks

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**